src/test/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActorTest.scala

```diff
@@ -302,6 +302,30 @@ class InstanceTrackerActorTest extends AkkaUnitTest {
     probe.expectMsg(appDataMap)
   }
 
+  "acknowledges and replies to the correct actors" in {
```

Comment:
I just added a test because I wanted to be sure this really works as weirdly as it does now... The current flow is

Delegate -? InstanceTracker -> UpdateActor -? InstanceTracker 

And from the last ask call to InstanceTracker we mark as finished the ask query from delegate as well as UpdateActor...

I think that UpdateActor should ACK by itself when the update operation finishes (InstanceTracker processes the update and reply).

I will do this in another PR...

Reply:
I might be wrong, but I think this can be detangled even further (in that follow-up PR you mentioned). Questions that might help detangling:

* What's the main purpose of the InstanceUpdateActor?
* What is needed to resolve the effect of a given operation? (hint: the current state, loaded from … the instanceTracker)
* What does the InstanceUpdateActor do, apart from putting updated into a byInstance-Queue?
* What would prevent quite some read requests (routed via the InstanceTrackerDelegate)?

Reply:
fair points (noted). In the next iteration I would like to get rid of the Ack class, but there's much more simplifications that can be done on this front, there it lot of back and forth communication for sure as you said...

And yes the instanceupdateactor really does not do anything else than making sure updates of the same instance does not happen in parallel, maybe we can get rid of this guy altogether and use streams instead. Will have to think about it...